### PR TITLE
Change pushHash with stringified data

### PR DESCRIPTION
**Pull request by alenoir**
Notification with `loc-key` and `loc-args` crash because `data.alert` is not a string.

e.g. 

```js
alert: {
  'loc-key': '....',
  'loc-args': '...',
}
```

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
@alenoir updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@alenoir updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@alenoir updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@alenoir updated the pull request.

**Comment by alenoir** [comment]
as we said in https://github.com/ParsePlatform/parse-server/commit/676d2e28de1c335448d1c1a997371b011f9630e2 i replace payloadstring by stringify data.alert for localization format

**Comment by protspace** [comment]
the same for me. 
`info: warning: error while sending push TypeError: Not a string or buffer`
and then

`Error: {"code":141,"message":{"code":107,"message":"Received an error with invalid JSON from Parse: <html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr><center>nginx/1.6.2</center>\r\n</body>\r\n</html>\r\n"}}`

**Comment by kenkuan** [comment]
This should be merged soon! All iOS localized push will fail on version `2.2.17`...

File: src/pushStatusHandler.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ export default function pushStatusHandler(config) {
       expiry: body.expiration_time,
       status: "pending",
       numSent: 0,
-      pushHash: md5Hash(data.alert || ''),
+      pushHash: md5Hash(JSON.stringify(data.alert) || ''),
```
Comment (flovilmart) [inline src/pushStatusHandler.js]: in the case alert is a string, that will wrap the string into another string, including the quotes...
This should be updated to:

```
let pushHash;
if (typeof data.alert === 'string') {
  pushHash = md5Hash(data.alert);
} else if (typeof data.alert === 'object') {
  pushHash = md5Hash(JSON.stringify(data.alert));
} else {
   pushHash = "d41d8cd98f00b204e9800998ecf8427e"; // empty string hash
}
```

**Comment by flovilmart** [comment]
thanks @alenoir. Tests would have been nice too.

**Comment by facebook-github-bot-1** [comment]
@alenoir updated the pull request.

**Comment by flovilmart** [comment]
Thanks @alenoir, merging after travis

**Pull request merged**